display_lang() to return user's language, if set.

kaapstorm opened the pull request:
This addresses [FB 170191](http://manage.dimagi.com/default.asp?170191) where the names of modules and forms in report filter dropdowns are always given in English, instead of the language the HQ user has chosen.

Elven @dannyroberts, ping @snopoke

kaapstorm committed:
display_lang to return user's language, if set

Fall back to English.

snopoke commented:
:white_check_mark:

czue commented on corehq/apps/reports/filters/forms.py:
```diff
@@ -45,8 +45,7 @@ def display_lang(self):
         """
-        # todo make this functional
-        return 'en'
+        return self.request.couch_user.language or 'en'
```
lol, 2.5 years later....

czue replied:
do we ever have to worry about `self.request` or `self.request.couch_user` being `None`? might be safe to use `getattr`, or if you're convinced they're always set then feel free to merge as is

kaapstorm replied:
Good point, @czue. couch_user can be an instance of CouchUser, InvalidUser and PublicUser. InvalidUser and PublicUser both extend CouchUser, which has a language property. So it should always be there, even if it isn't set.

czue replied:
and I think you're implying this, but you're sure that `request` is always populated and `couch_user` is always one of those three classes on `request`. as long as that's right go ahead and self merge.

kaapstorm replied:
FormsByApplicationFilter extends BaseReportFilter, and `request` is a mandatory parameter, so we know it will always be set. I'm pretty sure all reports are only available to authenticated users, but Just In Case, it's probably wise to check that couch_user is set. Thanks @czue.

kaapstorm committed:
Check for request.couch_user Just In Case

FormsByApplicationFilter extends BaseReportFilter, and request is a mandatory parameter, so we know it will always be set. I'm pretty sure reports are only available to authenticated users, but Just In Case, check couch_user is set.
